src/main/webapp/app/entities/participation/participation.service.ts

```diff
@@ -96,6 +96,37 @@ export class ParticipationService {
         });
     }
 
+    mergeProgrammingParticipations(participations: ProgrammingExerciseStudentParticipation[]): ProgrammingExerciseStudentParticipation {
+        const combinedParticipation = new ProgrammingExerciseStudentParticipation();
+        if (participations && participations.length > 0) {
+            combinedParticipation.repositoryUrl = participations[0].repositoryUrl;
+            combinedParticipation.initializationState = participations[0].initializationState;
+            combinedParticipation.initializationDate = participations[0].initializationDate;
+            combinedParticipation.presentationScore = participations[0].presentationScore;
+            combinedParticipation.buildPlanId = participations[0].buildPlanId;
+            this.mergeResultsAndSubmissions(combinedParticipation, participations);
+        }
+        return combinedParticipation;
+    }
+
+    mergeStudentParticipations(participations: StudentParticipation[]): StudentParticipation {
+        const combinedParticipation = new StudentParticipation();
+        if (participations && participations.length > 0) {
+            this.mergeResultsAndSubmissions(combinedParticipation, participations);
+        }
+        return combinedParticipation;
+    }
+
+    public convertParticipationsDateFromServer(participations: StudentParticipation[]) {
```

Comment:
Since you didn't change this method, can you move it back at it's original place?

Reply:
I have no idea how the order has changed. In any case, I have adapted it to the state from the develop branch. Thank you!